python_modules/dagster-graphql/dagster_graphql/schema/backfill.py

```diff
@@ -331,17 +339,54 @@ def _get_records(self, graphene_info: ResolveInfo) -> Sequence[RunRecord]:
 
     def _get_partition_run_data(self, graphene_info: ResolveInfo) -> Sequence[RunPartitionData]:
         if self._partition_run_data is None:
-            self._partition_run_data = (
-                graphene_info.context.instance.run_storage.get_run_partition_data(
-                    runs_filter=RunsFilter(
-                        tags={
-                            BACKFILL_ID_TAG: self._backfill_job.backfill_id,
-                        }
+            # Storage layer not equipped to calculate partition status for ranged job backfills so
+            # we perform the calculation ad hoc here.
+            partition_set = self._get_partition_set(graphene_info)
+            if (
+                partition_set
+                and partition_set.backfill_policy
+                and partition_set.backfill_policy.max_partitions_per_run != 1
+            ):  # ranged backfills
+                self._partition_run_data = self._get_partition_run_data_for_ranged_job_backfill(
+                    graphene_info.context.instance, partition_set
+                )
+            else:
+                self._partition_run_data = (
+                    graphene_info.context.instance.run_storage.get_run_partition_data(
+                        runs_filter=RunsFilter(
+                            tags={
+                                BACKFILL_ID_TAG: self._backfill_job.backfill_id,
+                            }
```

Comment:
Kind of a silly nit, but could this be changed to `DagsterRun.tags_for_backfill_id()` like 370? is one newer/better vs. the other?